src/main/java/brightspark/sparkshammers/integration/crafttweaker/CTHammerCraftingTableRecipes.java

```diff
@@ -0,0 +1,120 @@
+package brightspark.sparkshammers.integration.crafttweaker;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import brightspark.sparkshammers.hammerCrafting.HammerCraftingManager;
+import brightspark.sparkshammers.hammerCrafting.HammerShapedOreRecipe;
+import brightspark.sparkshammers.init.SHRecipes;
+import brightspark.sparkshammers.reference.Reference;
+import crafttweaker.CraftTweakerAPI;
+import crafttweaker.api.item.IItemStack;
+import crafttweaker.api.minecraft.CraftTweakerMC;
+import crafttweaker.mc1120.CraftTweaker;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.registries.IForgeRegistry;
+import net.minecraftforge.registries.IForgeRegistryModifiable;
+import net.minecraftforge.registries.RegistryBuilder;
+import stanhebben.zenscript.annotations.ZenClass;
+import stanhebben.zenscript.annotations.ZenMethod;
+
+@ZenClass("mods.sparkshammers.HammerCrafting")
+public class CTHammerCraftingTableRecipes {
+
+	public static class HammerCraftingAction extends AddRemoveAction {
+		
+		public HammerCraftingAction(ItemStack output, List<ItemStack> input) {
+			this.output = output;
+			this.input = input;
+		}
+		
+		ItemStack output;
+		
+		List<ItemStack> input;
+
+		/**
+		 * Adds a recipe.
+		 * Due to {@link net.minecraftforge.common.crafting.CraftingHelper} the ingredients
+		 * need to be parsed individually.
+		 */
+		@Override
+		protected void add() {
+			HammerCraftingManager.REGISTRY.register(new HammerShapedOreRecipe(output, 
+					"ABCDE", "FGHIJ", "KLMN ", 
+					'A', input.get(0),  'B', input.get(1),  'C', input.get(2),  'D', input.get(3), 'E', input.get(4),
+					'F', input.get(5),  'G', input.get(6),  'H', input.get(7),  'I', input.get(8), 'J', input.get(9),
+					'K', input.get(10), 'L', input.get(11), 'M', input.get(12), 'N', input.get(13)
+					));
+		}
+
+		
+		/**
+		 * Only looks up and removes a recipe based on the output
+		 */
+		@Override
+		protected void remove() {
+			HammerShapedOreRecipe wanted = null;
+			try {
+				wanted = HammerCraftingManager.getRecipes().stream().filter(recipe -> recipe.getRecipeOutput().isItemEqual(output)).findFirst().get();

```

Comment:
Would only remove the first recipe with that output, you might want do go for a `foreach` or similar

Reply:
Due to the registry's setup, it is impossible for more than 1 recipes to exist with equal outputs.
[HammerShapedOreRecipe:43](https://github.com/thebrightspark/SparksHammers/blob/1.12/src/main/java/brightspark/sparkshammers/hammerCrafting/HammerShapedOreRecipe.java#L43)